yggdrasil-utils/src/main/java/org/hyperagents/yggdrasil/utils/impl/RepresentationFactoryHMASImpl.java

```diff
@@ -94,418 +104,444 @@ private Signifier webSubSignifier(final WebSubSpecs specs,
+      final boolean isCartagoWorkspace
   ) {
     // TODO: Add artifactTemplates to makeArtifact signifier
     final String baseUri = this.httpConfig.getWorkspaceUri(workspaceName);
     final Workspace workspace = new Workspace.Builder()
-      .setIRIAsString(baseUri + "#workspace")
-      .addSemanticType(HMAS + "Workspace")
-      .build();
+        .setIRIAsString(baseUri + "#workspace")
+        .addSemanticType(HMAS + "Workspace")
+        .build();
 
     // makeArtifact Signifier
     final Form makeArtifactForm = new Form.Builder(baseUri + "artifacts/")
-      .setMethodName(HttpMethod.POST.name())
-      .setContentType("application/json")
-      .setIRIAsString(baseUri + "#makeArtifactForm")
-      .build();
+        .setMethodName(HttpMethod.POST.name())
+        .setContentType("application/json")
+        .setIRIAsString(baseUri + "#makeArtifactForm")
+        .build();
     // TODO: Add inputSpecification to makeArtifact
     final QualifiedValueSpecification makeArtifactInput = new QualifiedValueSpecification.Builder()
-      .addRequiredSemanticType(CORE.TERM.ARTIFACT.toString())
-      .setIRIAsString("http://example.org/artifact-shape")
-      .setRequired(true)
-      .addPropertySpecification(JACAMO + "hasName",
-        new StringSpecification.Builder()
+        .addRequiredSemanticType(CORE.TERM.ARTIFACT.toString())
+        .setIRIAsString("http://example.org/artifact-shape")
+        .setRequired(true)
+        .addPropertySpecification(JACAMO + "hasName",
+          new StringSpecification.Builder()
           .setRequired(true)
           .setName("Name")
           .setDescription("The name of the created artifact")
           .build())
-      .addPropertySpecification(JACAMO + "hasClass",
-        new StringSpecification.Builder()
+        .addPropertySpecification(JACAMO + "hasClass",
+          new StringSpecification.Builder()
           .setRequired(true)
           .setName("Class")
           .setDescription("The class of the created artifact")
           .build())
-      .addPropertySpecification(JACAMO + "hasInitialisationParameters",
-        new StringSpecification.Builder()
+        .addPropertySpecification(JACAMO + "hasInitialisationParameters",
+          new StringSpecification.Builder()
           .setName("Initialization parameters")
           .setDescription("A list containing the parameters for initializing the artifacts")
           .build())
-      .build();
+        .build();
 
     // registerArtifact Signifier
     final Form registerArtifactForm = new Form.Builder(baseUri + "artifacts/")
-      .setMethodName(HttpMethod.POST.name())
-      .setContentType(CONTENT_TYPE_TURTLE)
-      .setIRIAsString(baseUri + "#registerArtifactForm")
-      .build();
-    final QualifiedValueSpecification registerArtifactInput = new QualifiedValueSpecification.Builder()
-      .addRequiredSemanticType(CORE.TERM.ARTIFACT.toString())
-      .setIRIAsString("http://example.org/artifact-rdf")
```

Comment:
Consider replacing "http://example.org/artifact-rdf" with an IRI that uses the `baseUri`, e.g., `baseUri + "#artifact-rdf"`.